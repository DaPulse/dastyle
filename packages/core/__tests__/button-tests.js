import React from 'react';
import Button from '../src/components/button/button';
import renderer from 'react-test-renderer';

test('Button changes the class when hovered', () => {

  const component = renderer.create(
    <Button>Test Button</Button>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onMouseEnter();

  tree = component.toJSON();

  expect(tree).toMatchSnapshot();


  tree.props.onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onMouseDown();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
