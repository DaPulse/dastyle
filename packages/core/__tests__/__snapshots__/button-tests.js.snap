exports[`test Button changes the class when hovered 1`] = `
<button
  className="normal"
  href="#"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}>
  Test Button
</button>
`;

exports[`test Button changes the class when hovered 2`] = `
<button
  className="hovered"
  href="#"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}>
  Test Button
</button>
`;

exports[`test Button changes the class when hovered 3`] = `
<button
  className="normal"
  href="#"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}>
  Test Button
</button>
`;

exports[`test Button changes the class when hovered 4`] = `
<button
  className="down"
  href="#"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}>
  Test Button
</button>
`;
